go/types/map_test.go

```diff
@@ -1155,3 +1155,41 @@ func TestMapRemoveLastWhenNotLoaded(t *testing.T) {
 		assert.True(tm.toMap().Equals(nm))
 	}
 }
+
+func TestMapIterFrom(t *testing.T) {
+	assert := assert.New(t)
+
+	test := func(m Map, rangeStart, rangeEnd, expectedStart, expectedEnd Value) {
```

Comment:
Oh - one way to do that would be to have a function which flattens a Map into a ValueSlice using IterFrom, then use ValueSlice.Equals(mapKVs[some:range]).